src/metabase/models/setting.clj

```diff
@@ -1079,6 +1086,14 @@
        (not= (:setter setting-definition) :none)
        (not (ns-in-test? (:namespace setting-definition)))))
 
+(defn- inherit-base
```

Comment:
```suggestion
(defn- maybe-inherit-base
```

Reply:
When I see a function starting with "maybe" I assume this means it returns something optional, but in this case it always returns a map. That said, the "inherit" verb makes it clear that this is a transformation, so I guess it's clear that it means it is "maybe" a no-op in this case. I don't think the extra verbiage adds an important distinction though, in the same way that `merge` handles nil but isn't called `merge-maybe`. Let me know if you feel strongly.

Reply:
I'm with Chris on this one, especially since this is now called `merge-base` not `inherit-base`. I find it intuitive that if a base doesn't exist this is a no-op, like `(merge x nil)`.